3478.release docs

meejah opened the pull request:
This was written while making the 1.15.0-rc0 release-candidate. Although it still needs a couple more details, comments are welcome. This (or something like it) should be merged after the actual 1.15.0 release, probably.

sajith commented:
Turns out that zfec 1.5.4 is broken on Windows.  Nothing broke [when I tested it though](https://github.com/tahoe-lafs/tahoe-lafs/pull/804)... and I don't know what this is about.  I guess deleting the Windows wheel packages from PyPI should fix the CI problem here.  But my PyPI account do not have the permissions to do that.

sajith commented:
Deleted zfec's py27+windows .whl packages from PyPI and re-ran the CI.  I still need to find out why they are broken.

exarkun commented on docs/release-checklist.rst:
```diff
@@ -0,0 +1,155 @@
+- (original guide says only "twine upload dist/*")
+- the following developers have access to do this:
+  - exarkun
+  - warner
```
I wonder if I do.  PyPI says you, warner, and I are "maintainers" on the PyPI project.  I don't know what the "maintainer" permission confers.  I can't find any kind of administrative interface for the project so I can't see what permissions anyone has or grant any new permissions.

sajith replied:
What I learned last week is that a PyPI project maintainer can upload packages, but they cannot "manage" things (which includes permission to yank/delete a release or delete some packages selectively).  You have to be a project "owner" in order to be able to yank/delete stuff.

@vu3rdd made me a zfec project "owner", which is how I could delete those broken .whl packages.

exarkun commented on docs/release-checklist.rst:
```diff
@@ -0,0 +1,155 @@
+the dependencies. There are no instructions on how to collect these or
+where to put them on the tahoe-lafs.org machines.
+
+Is this step still useful?
```
I support dropping this one.

meejah replied:
Should be delete the stuff that's on tahoe-lafs.org now as well? Anyway I filed https://tahoe-lafs.org/trac/tahoe-lafs/ticket/3494#ticket

exarkun replied:
I guess there's a chance someone is operating an older release according to older installation instructions and could get mad if the tahoe-lafs.org-hosted artifacts vanished...

If it is real, that is an unfortunate situation.  Perhaps we can determine if it is real by inspecting the HTTP access logs on tahoe-lafs.org - if there are any.

Hm.  There are access logs and there are some hits on `/deps/` from something claiming to be pip.  So maybe no clear case to be made to delete that stuff right away.  Probably should update the install docs that refer to that URL and get that into a release first, at minimum.

exarkun reviewed (commented):
Thanks.  Left some comments inline.  By the end of the instructions I think we've gotten to finishing an rc0.  I guess there should be a loop in the instructions somewhere?

Also I see the rc0 isn't on PyPI right now - I suppose because you don't have upload permission?  I could try to upload it, if we want pre-releases on PyPI.

exarkun commented:
Another thing to do is add an item to the news list on the tahoe-lafs.org front page, eg https://tahoe-lafs.org/trac/tahoe-lafs/wiki/WikiStart?action=diff&version=582&old_version=581

meejah commented on docs/release-checklist.rst:
```diff
@@ -0,0 +1,155 @@
+The `master` branch should always be releasable. However, it is worth
+asking on appropriate channels (IRC, the mailing-list, Nuts and Bolts
+meetings) whether there are interesting changes that should be
+included (or NOT included) etc.
```
Yeah, I agree too. I guess there's some argument to ask in case there's an important change "just about" to land though?

exarkun replied:
Maybe communicating about release schedule addresses this equally well?  "The release will be cut Tuesday at noon, if you want to get any final things in."

meejah commented on docs/release-checklist.rst:
```diff
@@ -0,0 +1,155 @@
+  - newsfragments/<ticket number>.minor
+  - commit it
+- manually fix NEWS.txt
+  - proper title for lastest release (instead of "Release ...post1432")
```
"Release 1.15.0" (it'll say e.g. "1.14.0 port1432" if there are 1432 commits since 1.14.0 I think..)

meejah commented on docs/release-checklist.rst:
```diff
@@ -0,0 +1,155 @@
+  - commit it
+- manually fix NEWS.txt
+  - proper title for lastest release (instead of "Release ...post1432")
+  - double-check date
```
Maybe you don't plan to actually do the release on the same day you did "tox -e news".

meejah commented on docs/release-checklist.rst:
```diff
@@ -0,0 +1,155 @@
+  - proper title for lastest release (instead of "Release ...post1432")
+  - double-check date
+  - spot-check the release notes (these come from the newsfragments
+    files though so don't do heavy editing)
```
I guess because whomever wrote and reviewed a particular change presumably thought about what to write about it and someone else changing it all "as" the release happens seems .. bad?

meejah commented on docs/release-checklist.rst:
```diff
@@ -0,0 +1,155 @@
+
+- (all steps above are completed)
+- sign the release
+  - git tag -s -u 0xE34E62D06D0E69CFCA4179FFBDE0D31D68666A7A -m "release Tahoe-LAFS-1.15.0rc0" tahoe-lafs-1.15.0.rc0
```
hmm okay, so it should be "1.15.0rc0" then is what that warning means?

exarkun replied:
I *think* so.  https://www.python.org/dev/peps/pep-0440/#public-version-identifiers seems to be the canonical definition of the syntax and if am scrying the regex properly (one of my favorite activities) this supports the idea that there is no `.` separating the last numerical component from a "a", "b", or "rc" suffix.

meejah commented on docs/release-checklist.rst:
```diff
@@ -0,0 +1,155 @@
+  - ls dist/ | grep 1.15.0rc0
+- inspect and test the tarballs
+  - install each in a fresh virtualenv
+  - run basic tests
```
Whatever the release-person decides, I guess. We don't have any of those written down (and the closest we do have, the integration tests, should already have been run).

Do you think something should be written down here? What?

exarkun replied:
Nah I think passing automated test suite should be all that is required of the release manager.  If folks reading the announcement want to do some tests, awesome, that's probably the reason we're doing an rc after all.

So maybe just strike this item?

meejah commented:
okay, i think i addressed all the comments.

As for the PyPI upload: *do* we want release-candidates there? I don't know what pip / poetry / etc do with "rcX" things ...
